apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/QuestionCard.tsx

```diff
@@ -98,6 +98,23 @@ export default function QuestionCard({
   const question = localSurvey.questions[questionIdx];
   const open = activeQuestionId === question.id;
   const [openAdvanced, setOpenAdvanced] = useState(question.logic && question.logic.length > 0);
+  const [questionsWithEmptyLabel, setquestionsWithEmptyLabel] = useState<string[]>([]);
+
+  const editNextButtonLabel = (labelValue: string) => {
+    localSurvey.questions.forEach((q, index) => {
+      if (!q.buttonLabel || q.buttonLabel?.trim() === "" || questionsWithEmptyLabel.includes(q.id)) {
+        if (!questionsWithEmptyLabel.includes(q.id)) {
+          setquestionsWithEmptyLabel((prevArray) => [...prevArray, q.id]);
+        }
+        updateQuestion(index, { buttonLabel: labelValue });
+      } else {
+        updateQuestion(questionIdx, { buttonLabel: labelValue });
+      }
+    });
+  };
```

Comment:
please make sure to add an empty line between two functions to improve readability